src/openrpc/lifecycle.json

```diff
@@ -1,379 +1,626 @@
-          "params": [
-            {
-              "name": "value",
-              "value": {
-                "state": "suspended",
-                "previous": "inactive"
-              }
-            }
-          ]
-        }
-      ]
-    },
-    {
-      "name": "unloading",
-      "tags": [
-        {
-          "name": "notifier",
-          "x-event": "Lifecycle.onUnloading"
+            "name": "restore",
+            "tags": [
+                {
+                    "name": "notifier",
+                    "x-notifier": "Lifecycle.restore",
+                    "x-event": "Lifecycle.onRestore"
+                },
+                {
+                    "name": "capabilities",
+                    "x-uses": [
+                        "xrn:firebolt:capability:lifecycle:state"
+                    ]
+                }
+            ],
+            "summary": "Listen to the restoration event",
+            "params": [
+                {
+                    "name": "listen",
+                    "required": true,
+                    "schema": {
+                        "type": "boolean"
+                    }
+                }
+            ],
+            "result": {
+                "name": "value",
+                "schema": {
+                    "$ref": "#/components/schemas/LifecycleEvent"
+                }
+            },
+            "examples": [
+                {
+                    "name": "Default Example",
+                    "params": [
+                        {
+                            "name": "listen",
+                            "value": true
+                        }
+                    ],
+                    "result": {
+                        "name": "Default Result",
+                        "value": {
+                            "state": "suspended",
+                            "previous": "hibernated"
+                        }
+                    }
+                }
+            ]
         },
         {
-          "name": "capabilities",
-          "x-uses": ["xrn:firebolt:capability:lifecycle:state"]
-        }
-      ],
-      "summary": "Listen to the unloading event",
-      "params": [
-        {
-          "name": "value",
-          "schema": {
-            "$ref": "#/components/schemas/LifecycleEvent"
-          }
+            "name": "destroy",
+            "tags": [
+                {
+                    "name": "notifier",
+                    "x-notifier": "Lifecycle.destroy",
+                    "x-event": "Lifecycle.onDestroy"
+                },
+                {
+                    "name": "capabilities",
+                    "x-uses": [
+                        "xrn:firebolt:capability:lifecycle:state"
+                    ]
+                }
+            ],
+            "summary": "Listen to the destroy event \n\nHowever, this event is unreliable and app may not get this notification in the event of platform needs to kill the app quickly",
+            "params": [
```

Comment:
Notifiers don't have a "listen" param, that's for the subscriber, onDestroy.
The parameter here should be the destroy reason.